packages/langium/src/grammar/langium-grammar-code-actions.ts

```diff
@@ -96,6 +98,23 @@ export class LangiumGrammarCodeActionProvider implements CodeActionProvider {
         return undefined;
     }
 
+    private fixCrossRefSyntax(diagnostic: Diagnostic, document: LangiumDocument): CodeAction {
+        return {
+            title: 'Replace \'|\' with \':\'',
+            kind: CodeActionKind.QuickFix,
+            diagnostics: [diagnostic],
+            isPreferred: true,
+            edit: {
+                changes: {
+                    [document.textDocument.uri]: [{
+                        range: diagnostic.range,
+                        newText: document.textDocument.getText(diagnostic.range).replace(/\|/, ':')
```

Comment:
The diagnostic range might include comments. For example, when the code action is applied to the following cross-ref definition, only the comment is modified, but not the actual cross-ref token separator.
```
[Foo/*|*/|ID]
```
See my comment above for a solution. The CST subtree should give you sufficient information to find the exact position of the character that needs to be replaced.